bindings/alicloud/sls/sls.go

```diff
@@ -0,0 +1,140 @@
+package sls
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	sls "github.com/aliyun/aliyun-log-go-sdk"
+	"github.com/aliyun/aliyun-log-go-sdk/producer"
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+)
+
+type AliCloudSlsLogstorage struct {
+	logger   logger.Logger
+	producer *producer.Producer
+	metadata SlsLogstorageMetadata
+}
+
+type SlsLogstorageMetadata struct {
+	Endpoint        string `json:"endpoint"`
+	AccessKeyID     string `json:"accessKeyID"`
+	AccessKeySecret string `json:"accessKeySecret"`
+}
+
+type Callback struct {
+	s *AliCloudSlsLogstorage
+}
+
+// parse metadata field
+func (s *AliCloudSlsLogstorage) Init(metadata bindings.Metadata) error {
+	m, err := s.parseMeta(metadata)
+	if err != nil {
+		return err
+	}
+	s.metadata = *m
+	producerConfig := producer.GetDefaultProducerConfig()
+	// the config properties in the component yaml file
+	producerConfig.Endpoint = m.Endpoint
+	producerConfig.AccessKeyID = m.AccessKeyID
+	producerConfig.AccessKeySecret = m.AccessKeySecret
+	s.producer = producer.InitProducer(producerConfig)
+
+	s.startSlsProducer()
+	return nil
+}
+
+func (s *AliCloudSlsLogstorage) startSlsProducer() {
+	notifyChan := make(chan os.Signal, 1)
```

Comment:
What is this channel used for?

Reply:
this channel is used to test if could close the sis producer. remove it  now